spec/mixlib/shellout/windows_spec.rb

```diff
@@ -307,4 +307,88 @@ def self.with_command(cmd, filename: nil, search: false, directory: false, &exam
       end
     end
   end
+
+  context "#combine_args" do
+    let(:shell_out) { Mixlib::ShellOut.new }
+    subject { shell_out.send(:combine_args, *largs) }
+
+    def self.with_args(*args, &example)
+      context "with command #{args}" do
+        let(:largs) { args }
+        it(&example)
+      end
+    end
+
+    with_args("echo", "%PATH%") do
+      is_expected.to eql(%q{echo %PATH%})
+    end
+
+    with_args("echo %PATH%") do
+      is_expected.to eql(%q{echo %PATH%})
+    end
+
+    # Note carefully for the following that single quotes in ruby support '\\' as an escape sequence for a single
+    # literal backslash.  It is not mandatory to always use this since '\a' does not esacpe the 's' and is literally
+    # a backlash followed by an 'a'.  However, in the following all backslashes are escaped for consistency.  Otherwise
+    # it becomes prohibitively confusing to track when you need and do not need the escape the backslash (particularly
+    # when the literal string has a trailing backslash such that '\\' must be used instead of '\' which would escape
+    # the intended terminating single quote or %q{\} which escapes the terminating delimiter).
+
+    with_args("child.exe", "argument1", "argument 2", '\\some\\path with\\spaces') do
+      is_expected.to eql(%q{child.exe argument1 "argument 2" "\\some\\path with\\spaces"})
+    end
```

Comment:
it might be nice if, for consistency, you used one form of quoting everywhere. `%q{}`? Not sure what chef-style thinks of that.

Reply:
I would probably go the other way and replace `%q` with single quotes since I want to preserve `with_args` looking more like a shell_out args.

The inconsistency in single-quotes / double-quotes is entirely imposed by chefstyle.  I went with single quotes and then chefstyle ripped across it and "fixed" double-quotes where it was safe to do so.